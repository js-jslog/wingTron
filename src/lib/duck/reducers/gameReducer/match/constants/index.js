// @flow

import type { Match } from '../types'

export const EXAMPLE_MATCH = {
  field_width: 200,
  field_height: 200,
  matches: 10
}

