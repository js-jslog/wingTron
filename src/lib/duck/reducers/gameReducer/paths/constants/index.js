// @flow

import type { Coord } from '../types'
import type { Path } from '../types'
import type { NamedPath } from '../types'
import type { PathsById } from '../types'
import type { PathList } from '../types'
import type { Paths } from '../types'

const EXAMPLE_COORD: Coord = [ 150, 100 ]
const EXAMPLE_PATH: Path = [
  EXAMPLE_COORD,
  EXAMPLE_COORD
]
const EXAMPLE_NAMED_PATH0: NamedPath = {
  id: '0',
  path: EXAMPLE_PATH
}
const EXAMPLE_NAMED_PATH1: NamedPath = {
  id: '1',
  path: EXAMPLE_PATH
}
const EXAMPLE_PATHS_BY_ID: PathsById = {
  '0': EXAMPLE_NAMED_PATH0,
  '1': EXAMPLE_NAMED_PATH1
}
const EXAMPLE_PATH_LIST: PathList = [ '0', '1' ]
export const EXAMPLE_PATHS = {
  byId: EXAMPLE_PATHS_BY_ID,
  allIds: EXAMPLE_PATH_LIST
}
