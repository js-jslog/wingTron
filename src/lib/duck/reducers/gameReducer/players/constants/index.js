// @flow

import type { Player } from '../types'
import type { PlayersById } from '../types'
import type { PlayerList } from '../types'
import type { Players } from '../types'

const EXAMPLE_PLAYER0: Player = {
  id: '0',
  path: '0',
  direction: 0,
  turn_left_keycode: 37,
  turn_right_keycode: 39,
  colour: 'rgba(255,0,0, 0.5)',
}

const EXAMPLE_PLAYER1: Player = {
  id: '1',
  path: '1',
  direction: Math.PI,
  turn_left_keycode: 65,
  turn_right_keycode: 68,
  colour: 'rgba(0,0,255, 0.5)',
}

const EXAMPLE_PLAYERS_BY_ID: PlayersById = {
  '0': EXAMPLE_PLAYER0,
  '1': EXAMPLE_PLAYER1
}
const EXAMPLE_PLAYER_LIST: PlayerList = [ '0', '1' ]

export const EXAMPLE_PLAYERS: Players = {
  byId: EXAMPLE_PLAYERS_BY_ID,
  allIds: EXAMPLE_PLAYER_LIST
}
