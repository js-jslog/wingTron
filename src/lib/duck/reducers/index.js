import * as ActionTypes from '~/duck/types/'

const INITIAL_STATE = {
  initial: 'state',
}

export const reducer = (state_in=INITIAL_STATE, action) => {
  switch(action.type) {

    case ActionTypes.UPDATE_OPTIONS: {

      return { ...action.options }
    }

    case ActionTypes.ADD_PLAYER_TO_OPTIONS: {
      const { players } = state_in

      return {
        ...state_in,
        players: [ ...players, {} ]
      }
    }

    case ActionTypes.REMOVE_PLAYER_FROM_OPTIONS: {
      const { index } = action
      const players = [ ...state_in.players ]

      players.splice(index, 1)

      return {
        ...state_in,
        players
      }
    }

    default: return state_in
  }
}
