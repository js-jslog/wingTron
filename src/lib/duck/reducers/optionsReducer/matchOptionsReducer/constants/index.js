// @flow

import type { MatchOptions } from '../types'

export const INITIAL_STATE: MatchOptions = {
  field_width: '200',
  field_height: '200',
  matches: '10'
}
