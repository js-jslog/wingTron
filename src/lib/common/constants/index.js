// @flow

import type { GameMatch, Player } from '~/common/flow-types'

import { INITIAL_STATE as EXAMPLE_MATCH_OPTIONS_IMPORT } from '~/duck/reducers/optionsReducer/matchOptionsReducer/constants'
import { INITIAL_PLAYER1_OPTIONS as EXAMPLE_PLAYER1_OPTIONS_IMPORT } from '~/duck/reducers/optionsReducer/playerOptionsReducer/constants'
import { INITIAL_PLAYER2_OPTIONS as EXAMPLE_PLAYER2_OPTIONS_IMPORT } from '~/duck/reducers/optionsReducer/playerOptionsReducer/constants'
import { EXAMPLE_PLAYERS as EXAMPLE_PLAYERS_IMPORT } from '~/duck/reducers/gameReducer/players/constants'

export const EXAMPLE_MATCH_OPTIONS = EXAMPLE_MATCH_OPTIONS_IMPORT
export const EXAMPLE_PLAYER1_OPTIONS = EXAMPLE_PLAYER1_OPTIONS_IMPORT
export const EXAMPLE_PLAYER2_OPTIONS = EXAMPLE_PLAYER2_OPTIONS_IMPORT

export const EXAMPLE_OPTIONS = {
  match: { ...EXAMPLE_MATCH_OPTIONS },
  players: [
    { ...EXAMPLE_PLAYER1_OPTIONS },
    { ...EXAMPLE_PLAYER2_OPTIONS }
  ]
}

export const EXAMPLE_GAME_MATCH: GameMatch = {
  field_width: 200,
  field_height: 200,
  matches: 10
}

export const EXAMPLE_PLAYERS = EXAMPLE_PLAYERS_IMPORT

export const EXAMPLE_GAME = {
  match: { ...EXAMPLE_GAME_MATCH },
  players: EXAMPLE_PLAYERS
}

