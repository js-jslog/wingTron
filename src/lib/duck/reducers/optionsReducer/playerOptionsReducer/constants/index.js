// @flow

import type { PlayerOptions } from '~/common/flow-types'
import { EXAMPLE_PLAYER1_OPTIONS, EXAMPLE_PLAYER2_OPTIONS } from '~/common/constants'

export const INITIAL_STATE: Array<PlayerOptions> = [
  { ...EXAMPLE_PLAYER1_OPTIONS },
  { ...EXAMPLE_PLAYER2_OPTIONS }
]
