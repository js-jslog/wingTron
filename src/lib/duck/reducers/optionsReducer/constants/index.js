// @flow

import type { Options } from '../'
import { INITIAL_STATE as EXAMPLE_MATCH_OPTIONS } from '../matchOptionsReducer/constants'
import { INITIAL_STATE as EXAMPLE_PLAYER_OPTIONS } from '../playerOptionsReducer/constants'

export const EXAMPLE_OPTIONS: Options = {
  match: EXAMPLE_MATCH_OPTIONS,
  players: EXAMPLE_PLAYER_OPTIONS
}
