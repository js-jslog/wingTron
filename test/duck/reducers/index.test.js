// @flow

import { EXAMPLE_OPTIONS } from '~/common/constants'
import { rootReducer } from '~/duck/reducers'
import { createStore } from 'redux'

describe('the integration of all the reducers', () => {

  test('the default store state', () => {

    const store = createStore(rootReducer)
    const state = store.getState()

    expect(state).toMatchSnapshot()
  })

  test('the store after a game start', () => {

    const store = createStore(rootReducer)
    const state = store.getState()
    const action = {
      type: 'START_GAME_FROM_OPTIONS',
      options: state.options
    }
    store.dispatch(action)
    
    expect(store.getState()).toMatchSnapshot()
  })

  test('the store after a game is created and progressed a few steps', () => {

    const store = createStore(rootReducer)

    const start_game_action = {
      type: 'START_GAME_FROM_OPTIONS',
      options: store.getState().options
    }
    store.dispatch(start_game_action)

    const progress_paths_action = {
      type: 'PROGRESS_PLAYER_PATHS',
      players: store.getState().game.players
    }
    store.dispatch(progress_paths_action)
    progress_paths_action.players = store.getState().game.players
    store.dispatch(progress_paths_action)
    progress_paths_action.players = store.getState().game.players
    store.dispatch(progress_paths_action)
    progress_paths_action.players = store.getState().game.players
    store.dispatch(progress_paths_action)

    expect(store.getState()).toMatchSnapshot()
  })
})

