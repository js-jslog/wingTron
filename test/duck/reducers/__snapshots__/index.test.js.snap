// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the integration of all the reducers the default store state 1`] = `
Object {
  "game": Object {
    "match": null,
    "paths": null,
    "players": null,
  },
  "options": Object {
    "match": Object {
      "field_height": "200",
      "field_width": "200",
      "matches": "10",
    },
    "players": Array [
      Object {
        "colour": "rgba(255,0,0, 0.5)",
        "direction": "0",
        "start_coord_x": "151",
        "start_coord_y": "100",
        "turn_left_keycode": "37",
        "turn_right_keycode": "39",
      },
      Object {
        "colour": "rgba(0,0,255, 0.5)",
        "direction": "3.141592653589793",
        "start_coord_x": "149",
        "start_coord_y": "100",
        "turn_left_keycode": "65",
        "turn_right_keycode": "68",
      },
    ],
  },
}
`;

exports[`the integration of all the reducers the store after a game is created and progressed a few steps 1`] = `
Object {
  "game": Object {
    "match": Object {
      "field_height": 200,
      "field_width": 200,
      "matches": 10,
    },
    "paths": Object {
      "allIds": Array [
        "0",
        "1",
      ],
      "byId": Object {
        "0": Object {
          "id": "0",
          "path": Array [
            Array [
              155,
              100,
            ],
            Array [
              151,
              100,
            ],
          ],
        },
        "1": Object {
          "id": "1",
          "path": Array [
            Array [
              145,
              100,
            ],
            Array [
              149,
              100,
            ],
          ],
        },
      },
    },
    "players": Object {
      "allIds": Array [
        "0",
        "1",
      ],
      "byId": Object {
        "0": Object {
          "colour": "rgba(255,0,0, 0.5)",
          "direction": 0,
          "id": "0",
          "path": "0",
          "turn_left_keycode": 37,
          "turn_right_keycode": 39,
        },
        "1": Object {
          "colour": "rgba(0,0,255, 0.5)",
          "direction": 3.141592653589793,
          "id": "1",
          "path": "1",
          "turn_left_keycode": 65,
          "turn_right_keycode": 68,
        },
      },
    },
  },
  "options": Object {
    "match": Object {
      "field_height": "200",
      "field_width": "200",
      "matches": "10",
    },
    "players": Array [
      Object {
        "colour": "rgba(255,0,0, 0.5)",
        "direction": "0",
        "start_coord_x": "151",
        "start_coord_y": "100",
        "turn_left_keycode": "37",
        "turn_right_keycode": "39",
      },
      Object {
        "colour": "rgba(0,0,255, 0.5)",
        "direction": "3.141592653589793",
        "start_coord_x": "149",
        "start_coord_y": "100",
        "turn_left_keycode": "65",
        "turn_right_keycode": "68",
      },
    ],
  },
}
`;

exports[`the integration of all the reducers the store after a game start 1`] = `
Object {
  "game": Object {
    "match": Object {
      "field_height": 200,
      "field_width": 200,
      "matches": 10,
    },
    "paths": Object {
      "allIds": Array [
        "0",
        "1",
      ],
      "byId": Object {
        "0": Object {
          "id": "0",
          "path": Array [
            Array [
              151,
              100,
            ],
            Array [
              151,
              100,
            ],
          ],
        },
        "1": Object {
          "id": "1",
          "path": Array [
            Array [
              149,
              100,
            ],
            Array [
              149,
              100,
            ],
          ],
        },
      },
    },
    "players": Object {
      "allIds": Array [
        "0",
        "1",
      ],
      "byId": Object {
        "0": Object {
          "colour": "rgba(255,0,0, 0.5)",
          "direction": 0,
          "id": "0",
          "path": "0",
          "turn_left_keycode": 37,
          "turn_right_keycode": 39,
        },
        "1": Object {
          "colour": "rgba(0,0,255, 0.5)",
          "direction": 3.141592653589793,
          "id": "1",
          "path": "1",
          "turn_left_keycode": 65,
          "turn_right_keycode": 68,
        },
      },
    },
  },
  "options": Object {
    "match": Object {
      "field_height": "200",
      "field_width": "200",
      "matches": "10",
    },
    "players": Array [
      Object {
        "colour": "rgba(255,0,0, 0.5)",
        "direction": "0",
        "start_coord_x": "151",
        "start_coord_y": "100",
        "turn_left_keycode": "37",
        "turn_right_keycode": "39",
      },
      Object {
        "colour": "rgba(0,0,255, 0.5)",
        "direction": "3.141592653589793",
        "start_coord_x": "149",
        "start_coord_y": "100",
        "turn_left_keycode": "65",
        "turn_right_keycode": "68",
      },
    ],
  },
}
`;
